modules/Routing.js

```diff
@@ -28,6 +29,10 @@ function createRoute(element, parentRoute, namedRoutes) {
     options.willTransitionTo = createTransitionToHook(props.to, props.params, props.query);
     props.path = props.path || props.from || '*';
   } else {
+    invariant(
+      React.Children.count(props.children) || props.handler !== RouteHandler,
+      "A route without children must specify a handler that isn't RouteHandler"
```

Comment:
Let's say:

`RouteHandler may not be used as the handler for a route with no children`